Realm/RLMQueryUtil.mm

```diff
@@ -1850,6 +1855,31 @@ bool is_self_value_for_key_path_function_expression(NSExpression *expression)
                        @"Only support compound, comparison, and constant predicates");
     }
 }
+
+// This function returns the nested subscripts from a NSPredicate with the following format `anyCol[0]['key']['all']`
+// This will iterate each argument of the NSExpression and its nested NSExpression's and take the constant value
+// and create a PathElement for the query.
+void QueryBuilder::get_path_elements(std::vector<PathElement> &paths, NSExpression *expression) {
+    for (NSUInteger i = 0; i < expression.arguments.count; i++) {
+        if (expression.arguments[i].expressionType == NSFunctionExpressionType) {
+            get_path_elements(paths, expression.arguments[i]);
+        } else {
+            if (expression.arguments[i].expressionType == NSConstantValueExpressionType) {
```

Comment:
If this expression is any other type this just silently builds the wrong query? This function appears to be missing error checking.